public/pages/Dashboard/utils/utils.tsx

```diff
@@ -553,27 +564,21 @@ const buildDetectorAndIdMap = (
 
 const selectLatestDetetorIds = (
   orderedAnomalyData: object[],
-  needeDetectorNum: number
-): string[] => {
-  const uniqueIds = [
-    ...new Set(
-      orderedAnomalyData.map(anomalyData =>
-        get(anomalyData, AD_DOC_FIELDS.DETECTOR_ID, '')
-      )
-    ),
-  ];
-  if (uniqueIds.length <= needeDetectorNum) {
+  neededDetectorNum: number
+): Set<string> => {
+  const uniqueIds = new Set(
+    orderedAnomalyData.map(anomalyData =>
+      get(anomalyData, AD_DOC_FIELDS.DETECTOR_ID, '')
+    )
+  );
+  if (uniqueIds.size <= neededDetectorNum) {
     return uniqueIds;
   }
-  const latestDetectorIds = [] as string[];
+  const latestDetectorIds = new Set<string>();
   for (let anomalyData of orderedAnomalyData) {
-    if (
-      !latestDetectorIds.includes(
-        get(anomalyData, AD_DOC_FIELDS.DETECTOR_ID, '')
-      )
-    ) {
-    }
-    if (latestDetectorIds.length === needeDetectorNum) {
+    const detectorId = get(anomalyData, AD_DOC_FIELDS.DETECTOR_ID, '');
+    latestDetectorIds.add(detectorId);
+    if (latestDetectorIds.size === neededDetectorNum) {
       return latestDetectorIds;
     }
   }
```

Comment:
It looks like this unnecessarily loops through the `orderedAnomalyData` again. Can't we just slice the `uniqueIds` set to include up to `neededDetectorNum` ids? Something like `return Array.from(uniqueIds).slice(0, neededDetectorNum`.

Reply:
Good idea. Didn't know that the elements of a set can be iterated in insertion order. Previously I was thinking using the loop to get the 1st `neededDetectorNum` detectors with anomalies in an ordered array. Will change to such way